### ISSUE TITLE
Support Alibaba-NLP/gte-multilingual-base

### ISSUE THREAD

**Issue by sigridjineth** [huggingface/text-embeddings-inference#366]
### Model description

The GTE model family has consistent support on Hugging Face TEI, but the newly introduced `gte-multilingual-base` fails to be supported at this moment.

```
2024-08-03T08:44:51.913133Z  INFO download_pool_config: text_embeddings_core::download: core/src/download.rs:38: Downloading `1_Pooling/config.json`
2024-08-03T08:44:52.701241Z  INFO download_new_st_config: text_embeddings_core::download: core/src/download.rs:62: Downloading `config_sentence_transformers.json`
2024-08-03T08:44:52.885881Z  INFO download_artifacts: text_embeddings_core::download: core/src/download.rs:21: Starting download
2024-08-03T08:44:52.885905Z  INFO download_artifacts: text_embeddings_core::download: core/src/download.rs:23: Downloading `config.json`
2024-08-03T08:44:53.270526Z  INFO download_artifacts: text_embeddings_core::download: core/src/download.rs:26: Downloading `tokenizer.json`
2024-08-03T08:44:53.749799Z  INFO download_artifacts: text_embeddings_backend: backends/src/lib.rs:331: Downloading `model.safetensors`
2024-08-03T08:44:57.772619Z  INFO download_artifacts: text_embeddings_core::download: core/src/download.rs:32: Model artifacts downloaded in 4.886736258s
Error: `config.json` does not contain `id2label`
```

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

https://huggingface.co/Alibaba-NLP/gte-multilingual-base

**Comment by kozistr** [comment]
TEI determines the type of backend model by the `architectures` in `config.json` [code](https://github.com/huggingface/text-embeddings-inference/blob/9ec8f34204c2818de96a8f910a74550b95a60dd8/router/src/lib.rs#L363).

However, `gte-multilingual-base` architectures in `config.json` are `NewModel` and `NewForTokenClassification`. I guess it'd be good to change the config (e.g. removing `NewForTokenClassification` from the architectures)

the reranker has a correct architecture name, but the config does not contain `id2label`. So, I assume adding `id2label` seems good (or loading as a reranker model if there's no `id2label` key in the config.json).

In short, opening a PR to remove/add the config can solve the issues.

**Comment by sigridjineth** [comment]
@kozistr nice guidance. will look into it

**Comment by nbroad1881** [comment]
https://huggingface.co/Alibaba-NLP/gte-multilingual-base/discussions/7#66bfb82ea03b764ca92a2221

**Comment by sigridjineth** [comment]
@nbroad1881 yup, I've found that renaming weight keys to remove prefixes like "new" and ensuring they follow the standard model naming pattern (e.g., "encoder.layer.0.attention.self.query.weight") improves compatibility for my own,
